tests/x11regressions/gnomecase/gnome_default_applications.pm

```diff
@@ -20,23 +20,48 @@ sub run {
     x11_start_program("xterm");
     assert_screen 'xterm';
 
+    my @applications = (
+        ['image/jpg',           'eog.desktop'],
+        ['image/png',           'eog.desktop'],
+        ['application/pdf',     'evince.desktop'],
+        ['application/x-bzip2', 'org.gnome.FileRoller.desktop'],
+        ['application/gzip',    'org.gnome.FileRoller.desktop']);
+    my $defaultApps = check_default_apps(@applications);
+    if ($defaultApps) {
+        prepare_application_environment();
+        open_default_apps();
+        clear_application_environment();
+    }
+
+    # clear the xterm
+    send_key 'alt-f4';
+
+}
+
+sub clear_application_environment {
+    x11_start_program("rm -rf /home/$username/gnometest");    # Clean the test directory
+}
+
+sub prepare_application_environment {
     assert_script_run "mkdir gnometest";
     assert_script_run "wget -P /home/$username/gnometest " . autoinst_url . "/data/x11regressions/test.pdf";
     assert_script_run "wget -P /home/$username/gnometest " . autoinst_url . "/data/x11regressions/shotwell_test.jpg";
     assert_script_run "wget -P /home/$username/gnometest " . autoinst_url . "/data/x11regressions/shotwell_test.png";
     assert_script_run "cp /usr/share/w3m/w3mhelp.html /home/$username/gnometest/";
     assert_script_run "tar cjvf /home/$username/gnometest/test.tar.bz2 -C /home/$username/gnometest/ test.pdf";
     assert_script_run "tar czvf /home/$username/gnometest/test.tar.gz -C /home/$username/gnometest/ test.pdf";
-    send_key 'alt-f4';
 
     # Open nautilus
     x11_start_program("nautilus");
     assert_screen 'nautilus-launched', 3;
     send_key "ctrl-l";
     type_string "/home/$username/gnometest\n";
     send_key "ret";
+    sleep 5;
```

Comment:
Why should we need a sleep 5 in front of an assert_screen? I don't think that is needed. Did you hit a problem why you added this?